### PR TITLE
Add logging on async rule retry

### PR DESCRIPTION
**Pull request by lchanmann**
Log warning on exception before entering retry and log error when max retry has been reached.

### CODE REVIEW

File: src/foam/nanos/ruler/RetryManager.java [hunk 1]
```diff
@@ -39,6 +46,7 @@ public void run() {
               latch_.countDown();
             }
           } catch (Exception ex) {
+            exception_ = ex;
```
Comment (nanoscott) [inline src/foam/nanos/ruler/RetryManager.java:49]: you should only keep the first exception .. so how about: 
```suggestion
            exception_ = (exception_ == null) ? ex : exception_;
```

Comment (jlhughes) [reply src/foam/nanos/ruler/RetryManager.java:49]: yes, agree.

Comment (lchanmann) [reply src/foam/nanos/ruler/RetryManager.java:49]: The first exception is already log in the RuleEngine class. This is the last exception when we stop retrying.

Comment (jlhughes) [reply src/foam/nanos/ruler/RetryManager.java:49]: Then I'm ok with it.

**Review by jlhughes — APPROVED** [review]

File: src/foam/nanos/ruler/RetryManager.java (hunk 1 above)
Comment (nanoscott) [reply src/foam/nanos/ruler/RetryManager.java:49]: I'm ok with it too.

**Review by nanoscott — APPROVED** [review]

**Pull request merged**